src/webui/api/torrentscontroller.cpp

```diff
@@ -980,14 +981,54 @@ void TorrentsController::renameAction()
     const QString hash = params()["hash"];
     QString name = params()["name"].trimmed();
 
-    if (name.isEmpty())
+    bool renameContainingFolder = (params()["renameContainingFolder"] == "true");
+
+    if (!Utils::Fs::isValidFileSystemName(name, true))
         throw APIError(APIErrorType::Conflict, tr("Incorrect torrent name"));
 
     BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
     if (!torrent)
         throw APIError(APIErrorType::NotFound);
 
     name.replace(QRegularExpression("\r?\n|\r"), " ");
+
+    if (renameContainingFolder) {
+        const QString oldContainingPath = torrent->name();
+        QString newContainingPath = name;
+
+        qDebug("WebUI Rename: moving \"%s\" to \"%s\"", qUtf8Printable(oldContainingPath), qUtf8Printable(newContainingPath));
+
+        // Renaming the containing folder
+        if (oldContainingPath == newContainingPath) {
```

Comment:
are those code copied from elsewhere? IIRC they were old code and were replaced.